### PR TITLE
Search ranking

### PR DESCRIPTION
**Pull request by aswanson-nr**
# Summary

- Adds search ranking
  - Ranking is defined by a list of rules that check for things like exact string match, word match, substring match, etc.
  - These rules are defined in priority order
- Make eslint ignore generated gatsby types

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -76,6 +76,7 @@
     "eslint-plugin-markdown": "^2.2.1",
     "jest": "^26.4.0",
     "prettier": "2.2.1",
+    "ts-jest": "^26.4.0",
```
Comment (josephgregoryii) [inline package.json:79]: I noticed the test file in this PR is still javascript - is this for possible future development for ts-jest?

**Review by josephgregoryii — APPROVED** [review]
Code looks cleaaaaaannnnn. :shipit:

**Review by mickeyryan42 — APPROVED** [review]
This looks awesome! The only thing i can think of is whether we want a comment on the regex in searchRanking.ts.  It's pretty self explanatory given the fn name so I don't think its a big deal either way.  Giving it an approval

File: package.json (hunk 1 above)
Comment (aswanson-nr) [reply package.json:79]: This was needed to get jest to run tests that use `.ts` files, so it was required even if the jest files are `.js`.
We use it over in the [quickstarts repo](https://github.com/newrelic/newrelic-quickstarts/blob/main/utils/package.json#L47)

**Pull request merged**